gdnative-derive/src/native_script/property_args.rs

```diff
@@ -1,29 +1,59 @@
 use syn::spanned::Spanned;
 
 pub struct PropertyAttrArgs {
+    pub ty: syn::Type,
     pub path: Option<String>,
     pub default: Option<syn::Lit>,
     pub hint: Option<syn::Path>,
     pub before_get: Option<syn::Path>,
+    pub use_get: bool,
+    pub get: Option<syn::Path>,
+    pub get_ref: Option<syn::Path>,
     pub after_get: Option<syn::Path>,
     pub before_set: Option<syn::Path>,
+    pub use_set: bool,
+    pub set: Option<syn::Path>,
```

Comment:
Do we need `use_get` and `use_set`?

Or could they be methods checking whether the corresponding `Option` fields are occupied? (to avoid storing the same information twice)

Reply:
But when use `#[property(get)]`, I am not sure how to set the `get`'s value.
It seems that `""`  is not a valid value for `syn::Path`.

Reply:
Ah, I see. Maybe something like
```rs
enum PropertyAttribute {
   None,
   Default,              // #[get] or #[set]
   WithPath(syn::Path),  // #[get("path")] or #[set("path")]
}
```

It's generally easier to reason about if one "state" is encoded as one field, and it also makes immediately clear which combinations are valid.